### ISSUE TITLE
Multiname Bug

### ISSUE THREAD

**Issue by michaelcheers** [bridgedotnet/Bridge#637]
[Live Sample](http://live.bridge.net/#760f2b63922075ddf62b)

This occurs when you have named the same as a class.
Eg.

    enum Operator
    {
          Add
    }
    static class App
    {
         [Ready]
         public static void Main ()
         {
                var Operator = Operator.Add;
                Console.Log(Operator == Operator.Add);
         }
    }
Works in C#.

**Comment by vladsch** [comment]
For me, your test sample is not compiled in VS2013
The error: "Cannot use local variable 'Operator' before it is declared"

**Comment by DaniilVeriga** [comment]
It looks like the Live sample is not compiled as well. There is the following:

> // Line 0, Col 0 : Failed to compare two elements in the array.

Yes, the error message is confusing, but as far as I can know it means that the C# source code is not compiled. Meanwhile, we work on making error message more informative. But it is another issue.

As for building the code sample in Visual Studio, I can confirm that I have the same that Vladimir does.

**Comment by michaelcheers** [comment]
@vladsch @DaniilVeriga  I gave you invalid code, but I have some new valid code for this issue.
I've got a dotnetfiddle sample this time.
[Brige Code](http://live.bridge.net/#c749bc3fcbe125cf49d75cb6be29016f)
[Dot Net Fiddle](https://dotnetfiddle.net/3vFWOv)
It should say true.

```csharp
/// <summary>
/// Creates a simple init class for our Application
/// </summary>
public enum Operator
{
    Add
}
public static class App
{
    // Use the [Ready] attribute to automaically
    // run this method when the page is ready.
    [Ready]
    public static void Main()
    {
        var Operator = (Operator)0;
        Console.Log(Operator.Add == Operator);
    }
}
```

**Comment by DaniilVeriga** [comment]
@michaelcheers, thank you for the test case! It was fixed in 1.12.1.
